### DISCUSSION TITLE
Milvus-MinIO bucket configuration programmatically

### DISCUSSION THREAD

**Discussion by UshaPravinkumar** [milvus-io/milvus#30751] [Q&A and General discussion]
Hello,
minio.bucketName - The name of the bucket where Milvus stores data in MinIO.
I have configured this setup manually and working smoothly.
Is there any chance to modify this bucket name in the configuration file programmatically?

**Comment by yhmo** [comment]
Milvus has no ability to rename a S3 bucket. It always accesses the S3 bucket which is specified by minio.bucketName.

You can use S3 sdk to rename the bucket by yourself, then change the mini.bucketName to the new name, and restart the milvus server.

**Reply by UshaPravinkumar** [reply]
No, I'm not renaming the bucket. For understanding, I have 2 buckets namely bucketA and bucketB in MinIO storage.
I have assigned bucketA for minio.bucketName attribute in Milvus.
If I want to change this to bucketB, I'm able to do this activity manually by editing the yaml file.
Is there any option to update this using any API or SDK?

**Reply by yhmo** [reply]
No such API.

Just remind:
1. milvus stores system metadata in Etcd, and stores data files in S3.
2. System metadata includes the S3 path of each segment/partition/collection, the metadata must be strongly consistent with the data files in S3. If you change the bucket to another, milvus will not be able to read the data files according to the path which is stored in Etcd.

**Reply by UshaPravinkumar** [reply]
Thanks for the details
